Replace `maybe_regular_expander` with `maybe_regular_expander_graph`

Peiffap opened the pull request:
This adds the `_graph` suffix to the function name. I hope I got all the places the old function was used!

_Follow-up to a comment by @dschult in https://github.com/networkx/networkx/pull/8048#pullrequestreview-2847815247_

amcandio reviewed (approved):
Would be cool to have a test that checks all returns graph methods are suffixed with "graph"

rossbar commented on networkx/generators/expanders.py:
```diff
@@ -324,13 +325,34 @@ def maybe_regular_expander(n, d, *, create_using=None, max_tries=100, seed=None)
+    import warnings
+
+    warnings.warn(
+        "maybe_regular_expander is deprecated,"
```
```suggestion
        "maybe_regular_expander is deprecated, "
```

rossbar reviewed (approved):
I'm +1 for this renaming - IMO the added clarity from the `_graph` suffix is worth the deprecation cycle!

I think this will also need an additional warnings filter in `conftest.py` in addition to the merge conflict resolution, but LGTM - thanks @Peiffap !

Peiffap commented on networkx/generators/expanders.py:
```diff
@@ -324,13 +325,34 @@ def maybe_regular_expander(n, d, *, create_using=None, max_tries=100, seed=None)
 
+def maybe_regular_expander(n, d, *, create_using=None, max_tries=100, seed=None):
+    """
+    .. deprecated:: 3.5
```
```suggestion
    .. deprecated:: 3.6
```

Peiffap commented:
Thanks for the review, @rossbar. I resolved the merge conflict (and moved the deprecation to 3.6), added your spacing suggestion, and added a filter for the warning.
